### ISSUE TITLE
Check for shema upgrade after checking of entiry existence

### ISSUE THREAD

**Issue by alyapunov** [tarantool/tarantool#8137]
After #7149 tarantool denies create/drop/alter of space/index/sequence/func if schema needs upgrade.

On the other hand at least create/drop functions have special options - `if_not_exists`/`if_exists` - which if are `true` in certain case actually make corresponding functions _not to do_ create/drop. However in this case those functions still throw an error:
```lua
tarantool> box.space.test ~= nil
---
- true
...

tarantool> box.schema.space.create('test', {if_not_exists = true})
---
- error: 'DDL operations are not allowed: Your schema version is 2.10.0 while Tarantool
    2.11.0-entrypoint-850-g182034dc9 requires a more recent schema version. Please,
    consider using box.schema.upgrade().'
...
```
We should check schema upgrade only if there is actual DDL.

One thing that must be notices that there are some cases when it's not possible even to check an entity existence without schema upgrade. For example if schema version is less than 1.7.5 there are no spaces such as _vspace, _vindex etc, which could be used for existence check. If so, we should throw 'DDL operations are not allowed' anyway.

**Comment by unera** [comment]
In 1-1 meeting, we created a (pre) list of possible if_not_exists places:

- creating space
- creating index
- dropping space
- creating user
- granting user

**Comment by Gumix** [comment]
This task is about a fix for #7149, but that patch is currently reverted.

**Comment by locker** [comment]
The issue was fixed by #8213.
